libfreerdp/codec/xcrush.c

```diff
@@ -732,13 +732,26 @@ static int xcrush_generate_output(XCRUSH_CONTEXT* xcrush, BYTE* OutputBuffer, UI
 	return 1;
 }
 
-static size_t xcrush_copy_bytes(BYTE* dst, const BYTE* src, size_t num)
+static INLINE size_t xcrush_copy_bytes(BYTE* dst, const BYTE* src, size_t num)
 {
-	size_t index;
-
-	for (index = 0; index < num; index++)
+	if (src + num < dst || src > dst + num)
+	{
+		memcpy(dst, src, num);
+	}
+	else
 	{
-		dst[index] = src[index];
+		// src and dst overlaps
+		// we should copy the area that doesn't overlap repeatly
+		size_t diff = (dst > src) ? dst - src : src - dst;
+		size_t rest = num % diff;
+		size_t end = num - rest;
+		for (size_t a = 0; a < end; a += diff)
```

Comment:
sorry, that is `C++` style and breaks with older compilers.

Reply:
I have put the declarations on the beginning of the function.